File: IvanShurenkov/Second Term/Block4/Bash/lib/bash/src/main/kotlin/lib/pipe/commands/PwdCommand.kt
package lib.pipe.commands

import lib.pipe.Output
import java.io.File
import java.nio.file.Paths

class PwdCommand(val name: String) : ICommand {
    override fun run(args: Array<String>, input: String?): Output {
        val path = Paths.get("").toRealPath().toString()

        var ls = ""
        val folders = File(path).listFiles { file -> file.isDirectory }
        folders?.forEach { folder -> ls += "${folder.name}/${System.lineSeparator()}" }
        val files = File(path).listFiles { file -> file.isFile }
        files?.forEach { file -> ls += "${file.name}${System.lineSeparator()}" }
        return Output("$path${System.lineSeparator()}$ls")
    }
}

File: Anton-Kraev/Term2/Block4/Bash/Bash/Commands/PwdCommand.cs
using System.Linq;

namespace Bash.Commands
{
    public class PwdCommand : ICommand
    {
        public string Name => "pwd";

        public string Execute(string args)
        {
            var dir = Directory.GetCurrentDirectory();
            var result = dir;
            var files = Directory.GetFiles(dir);
            foreach (var file in files)
            {
                result += "\n" + file;
            }
            return result;
        }
    }
}